src/server/routes/editor.js

```diff
@@ -250,47 +251,92 @@ router.get('/:id', (req, res, next) => {
+
+	const formattedRevisions = await Promise.all(
+		revisionsJSON.map(async rev => {
+			const note = await new Note()
+				.query('where', 'revision_id', '=', parseInt(rev.id, 10))
+				.fetchAll({require: false});
+			const noteJSON = note.toJSON();
+			const noteContent = noteJSON.map(currentNote => currentNote.content);
+
+			const {id: revisionId, ...otherProps} = rev;
+			return {entities: [], noteContent, revisionId, ...otherProps};
+		}));
+
+
+	const orderedRevisions = await utilis.getAssociatedEntityRevisions(formattedRevisions, req.app.locals.orm);
+
+	return orderedRevisions;
+}
+
+router.get('/:id/revisions', async (req, res, next) => {
 	const {Editor, TitleUnlock} = req.app.locals.orm;
-	new Editor({id: parseInt(req.params.id, 10)})
-		.fetch({
-			require: true,
-			withRelated: {
-				revisions(query) {
-					query.orderBy('id');
-				}
-			}
-		})
-		.then((editor) => getEditorTitleJSON(editor.toJSON(), TitleUnlock))
-		.then((editorJSON) => {
-			const props = generateProps(req, res, {
-				editor: editorJSON,
-				tabActive: 1
-			});
-			const markup = ReactDOMServer.renderToString(
-				<Layout {...propHelpers.extractLayoutProps(props)}>
-					<EditorContainer
-						{...propHelpers.extractEditorProps(props)}
-					>
-						<RevisionsTab
-							editor={props.editor}
-						/>
-					</EditorContainer>
-				</Layout>
-			);
-			res.send(target({
-				markup,
-				page: 'revisions',
-				props: escapeProps(props),
-				script: '/js/editor/editor.js'
-			}));
-		})
-		.catch(Editor.NotFoundError, () => {
-			throw new error.NotFoundError('Editor not found', req);
-		})
-		.catch(next);
+
+	const size = req.query.size ? parseInt(req.query.size, 10) : 20;
+	const from = req.query.from ? parseInt(req.query.from, 10) : 0;
+
+	const orderedRevisions = await getOrderedRevisionForEditorPage(from, size, req);
+
+	const editor = await new Editor({id: req.params.id}).fetch();
+	const editorJSON = await getEditorTitleJSON(editor.toJSON(), TitleUnlock);
+
+	const props = generateProps(req, res, {
+		editor: editorJSON,
+		from,
+		results: orderedRevisions,
+		showRevisionNote: true,
+		size,
+		tabActive: 1,
+		tableHeading: 'Revision History'
+	});
+
+	const markup = ReactDOMServer.renderToString(
+		<Layout {...propHelpers.extractLayoutProps(props)}>
+			<EditorContainer
+				{...propHelpers.extractEditorProps(props)}
+			>
+				<EditorRevisionPage
+					{...propHelpers.extractChildProps(props)}
+				/>
+			</EditorContainer>
+		</Layout>
+	);
+
+	res.send(target({
+		markup,
+		page: 'revisions',
+		props: escapeProps(props),
+		script: '/js/editor/editor.js'
+	}));
 });
 
+// eslint-disable-next-line consistent-return
+router.get('/:id/revisions/revisions', async (req, res, next) => {
+	const size = req.query.size ? parseInt(req.query.size, 10) : 20;
+	const from = req.query.from ? parseInt(req.query.from, 10) : 0;
+
+	try {
+		const orderedRevisions = await getOrderedRevisionForEditorPage(from, size, req);
```

Comment:
Here as well you should check first that the Editor exists, and return a `new error.NotFoundError('Editor not found', req);` otherwise.
At the moment visiting for a non-existing user (http://localhost:9099/editor/12789345/revisions/revisions)  I get an empty array, which is better that a random error, but I would expect a 404.

I think it's better to change check for the Editor in `getOrderedRevisionForEditorPage` then.

Again, this will be useful for reusing these tools for the API.